src/cell/unsafe_cell.rs

```diff
@@ -200,6 +215,21 @@ impl<T: ?Sized> UnsafeCell<T> {
             ptr: self.data.get(),
         }
     }
+
+    /// Get a mutable reference to the wrapped value.
+    ///
+    /// This function returns a [`MutRef`] guard, which is analogous to a
+    /// `&mut T`, but tracked by Loom. As long as the returned `MutRef`
+    /// exists, Loom will consider the cell to be accessed mutably.
+    ///
+    /// This is analogous to [`core::cell::UnsafeCell::get_mut`].
+    #[track_caller]
+    pub fn as_mut(&mut self) -> MutRef<'_, T> {
```

Comment:
I'm not entirely certain we even need tracking: if we can mutably borrow the value, we already have exclusive access to it, right? And because we borrow it mutably, no one can re-borrow it immutably until the borrow is dropped.

I've added the in-between struct so that someone can double-check my thoughts, but would be great to get some external input on it.